### PR TITLE
Improve CodeblocksOverFileUpload handler and refactor DirectoryUtils

### LINKED ISSUE #78: File upload detection is case sensitive
https://github.com/codesupport/discord-bot/blob/24e71cc2ea6af91aa6e7216a693d76cca1b6156c/src/event/handlers/CodeblocksOverFileUploadsHandler.ts#L15

### LINKED ISSUE #79: Add attempted upload file type to message for reference
https://github.com/codesupport/discord-bot/blob/24e71cc2ea6af91aa6e7216a693d76cca1b6156c/src/event/handlers/CodeblocksOverFileUploadsHandler.ts#L24

### PR DESCRIPTION
**Pull request by ajsaraujo**
Resolves #78, resolves #79

### CODE REVIEW

File: src/event/handlers/CodeblocksOverFileUploadsHandler.ts [hunk 1]
```diff
@@ -9,10 +9,14 @@ class CodeblocksOverFileUploadsHandler extends EventHandler {
...
 		let invalidFileFlag = false;
+		let invalidFileExtension: string = "";
 
 		if (message.attachments.size > 0) {
 			message.attachments.forEach(attachment => {
-				if (!ALLOWED_FILE_EXTENSIONS.includes(attachment.name?.split(".").pop()!)) {
+				const fileExtension = attachment.name?.split(".").pop()!.toLowerCase();
+
+				if (!ALLOWED_FILE_EXTENSIONS.includes(fileExtension)) {
+					invalidFileExtension = fileExtension;
```
Comment (ajsaraujo) [inline src/event/handlers/CodeblocksOverFileUploadsHandler.ts:19]: This assumes that only one file attachment can be sent per message, although we have a map of attachments... Anyway, I didn't find a way of sending a message with more than one file attachment yet.

File: test/event/handlers/CodeblocksOverFileUploadsHandlerTest.ts [hunk 2]
```diff
@@ -41,15 +40,23 @@ describe("CodeblocksOverFileUploadsHandler", () => {
 
 			expect(addMock.calledOnce).to.be.false;
 		});
-		
+
 		it("does nothing when there is a valid attachment.", async () => {
 			message.attachments.set("720390958847361064", new MessageAttachment("720390958847361064", "test.png"));
```
Comment (ajsaraujo) [inline test/event/handlers/CodeblocksOverFileUploadsHandlerTest.ts:45]: I thought that maybe we could have that string as a constant at the top of the file?

File: src/utils/DirectoryUtils.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    private static require = require;
+
+    static async getFilesInDirectory(directory: string, ending: string): Promise<any[]> {
+	    const directoryContents = await this.readDirectory(directory);
+
+	    /* eslint-disable */
+        return directoryContents
+            .filter(file => file.endsWith(ending))
+            .map(file => this.require(`${directory}/${file}`));
+        /* eslint-enable */
```
Comment (LamboCreeper) [inline src/utils/DirectoryUtils.ts]: Formatting seems a bit off.

File: src/utils/DirectoryUtils.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { readdir } from "fs";
+import { promisify } from "util";
+
+class DirectoryUtils {
+    static readDirectory = promisify(readdir);
+    private static require = require;
```
Comment (LamboCreeper) [inline src/utils/DirectoryUtils.ts]: Would like a blank line between these since they're sperate.

File: test/utils/DirectoryUtilsTest.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+		let sandbox: SinonSandbox;
+
+		beforeEach(() => {
+			sandbox = createSandbox();
+		});
+
+		afterEach(() => {
+			sandbox.restore();
+		});
+	});
```
Comment (LamboCreeper) [inline test/utils/DirectoryUtilsTest.ts]: Seems pointless?

**Review by LamboCreeper — CHANGES_REQUESTED** [review]

File: src/utils/DirectoryUtils.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+    static async getFilesInDirectory(directory: string, ending: string): Promise<any[]> {
+        const directoryContents = await this.readDirectory(directory);
+
+        /* eslint-disable */
+        return directoryContents
+            .filter(file => file.endsWith(ending))
+            .map(file => this.require(`${directory}/${file}`));
+        /* eslint-enable */
+    }
```
Comment (ajsaraujo) [inline src/utils/DirectoryUtils.ts:16]: The linter complains: `Expected indentation of 5 tabs, found 8 spaces`. 
However it's rendering fine here and on my editor. If I try to quick fix it I still get `Mixed indentation of tabs and spaces`, which renders awfully.

Comment (ajsaraujo) [reply src/utils/DirectoryUtils.ts:16]: I disabled the linter for that part of the code. :p

**Review by LamboCreeper — APPROVED** [review]

**Pull request merged**